app/src/main/java/org/dicekeys/app/fragments/PreferencesFragment.kt

```diff
@@ -83,10 +79,10 @@ class PreferencesFragment : PreferenceFragmentCompat(), SharedPreferences.OnShar
                         try {
                             val fd: FileDescriptor = fileDescriptor.fileDescriptor
                             val fileStream = FileInputStream(fd)
-                            val recipes: BackupRecipes = Json.decodeFromStream(fileStream) // Decode file contents into BackupRecipes
-                            recipeRepository.save(recipes.recipes)
+                            val recipes: List<BackupRecipe> = Json.decodeFromStream(fileStream) // Decode file contents into BackupRecipes
+                            recipeRepository.save(recipes.toDerivationRecipes())
```

Comment:
This looks like it will overwrite any existing recipes.  It should merge, eliminating duplicates that share the same type and recipeJson (keeping either the name from before the import or after.  what matters is that the other two fields are always preserved.)

Reply:
Recipes are saved by their id*, recipes with the same id are overwritten. All other recipes stay intact. So this works as a merge.

*id of a recipe is the hashCode of the recipeJson string. this give us a unique hash value of the recipe. But this is not correct, i should have used `type.hashCode + recipeJson.hashCode` to get the unique id of a recipe. I'll make a migration for this.

Reply:
Got it.  Please document to make it clear that `.save` is really `.add` and does not overwrite the old list.